src/hdf5/include/H5Epublic.h

```diff
@@ -52,8 +52,14 @@ typedef struct H5E_error2_t {
 #endif  /* _H5private_H */
 
 /* HDF5 error class */
+#ifdef __cplusplus
+extern "C" {
+#endif
 #define H5E_ERR_CLS		(H5OPEN H5E_ERR_CLS_g)
 H5_DLLVAR hid_t H5E_ERR_CLS_g;
+#ifdef __cplusplus
+}
+#endif

```

Comment:
Issue here is that it won't build with external headers, is there another way to handle this without a patch

Reply:
It will not build only under windows. Under linux and Mac the patch is not needed.